### PR TITLE
add dockerfile

### LINKED ISSUE #203: Release: add Docker image
To simplify a new user's experience, it would be useful to have a Docker image with gitbase pushed somewhere, as one of the release artefacts of the project.

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #203 

Publishing with CI will be done on another PR.

Final image is not alpine because we're using cgo and I could not get it to work in any way. If someone knows how to run CGO binaries on alpine, please come forward.

### CODE REVIEW

**Comment by jfontan** [comment]
We can revisit the base image of it in the future if needed but it's probably ok as is for now.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+RUN cd vendor/gopkg.in/bblfsh/client-go.v2 && make dependencies
+RUN go install -v github.com/src-d/gitbase/...
+
+FROM ubuntu:16.04
+
+COPY --from=builder /go/bin/gitbase /bin
+RUN mkdir -p /opt/repos
+
+ENV GITBASE_USER=gitbase
+ENV GITBASE_PASSWORD=""
```
Comment (kuba--) [inline Dockerfile]: just because I'm curious - do we rely on some image where `GITBASE_PASSWORD` is exported or we're just using empty password?

Comment (erizocosmico) [reply Dockerfile]: We're using no password by default and then you can add your own password when you run it using `-e GITBASE_PASSWORD="your password"`

**Comment by erizocosmico** [comment]
@jfontan yep, probably in the future we could run it as non-root user, as non pid 1 and so on

**Comment by erizocosmico** [comment]
Added tini to run gitbase as non pid 1

**Pull request merged**